integrationtests/helmops/controller/controller_test.go

```diff
@@ -0,0 +1,1006 @@
+		err := k8sClient.Get(ctx, types.NamespacedName{Name: helmapp.Name, Namespace: helmapp.Namespace}, &helmAppFromCluster)
+		if err != nil {
+			return err
+		}
+		helmAppFromCluster.Spec = helmapp.Spec
+		return k8sClient.Update(ctx, &helmAppFromCluster)
+	})
+}
+
+func getCondition(fllethelm *fleet.HelmApp, condType string) (genericcondition.GenericCondition, bool) {
+	for _, cond := range fllethelm.Status.Conditions {
+		if cond.Type == condType {
+			return cond, true
+		}
+	}
+	return genericcondition.GenericCondition{}, false
+}
+
+func checkConditionContains(g Gomega, fllethelm *fleet.HelmApp, condType string, status corev1.ConditionStatus, message string) {
+	cond, found := getCondition(fllethelm, condType)
+	g.Expect(found).To(BeTrue())
+	g.Expect(cond.Type).To(Equal(condType))
+	g.Expect(cond.Status).To(Equal(status))
+	g.Expect(cond.Message).To(ContainSubstring(message))
+}
+
+func newTLSServerWithAuth() *httptest.Server {
+	srv := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		username, password, ok := r.BasicAuth()
+		if ok {
+			usernameHash := sha256.Sum256([]byte(username))
+			passwordHash := sha256.Sum256([]byte(password))
+			expectedUsernameHash := sha256.Sum256([]byte(authUsername))
+			expectedPasswordHash := sha256.Sum256([]byte(authUsername))
+
+			usernameMatch := (subtle.ConstantTimeCompare(usernameHash[:], expectedUsernameHash[:]) == 1)
+			passwordMatch := (subtle.ConstantTimeCompare(passwordHash[:], expectedPasswordHash[:]) == 1)
+
+			if usernameMatch && passwordMatch {
+				w.WriteHeader(http.StatusOK)
+				fmt.Fprint(w, helmRepoIndex)
+			}
+		}
+
+		w.Header().Set("WWW-Authenticate", `Basic realm="restricted", charset="UTF-8"`)
+		http.Error(w, "Unauthorized", http.StatusUnauthorized)
+	}))
+	return srv
+}
+
+func getNewCustomTLSServer(handler http.Handler) (*httptest.Server, error) {
+	ts := httptest.NewUnstartedServer(handler)
+	serverCert, err := os.ReadFile("assets/server.crt")
+	if err != nil {
+		return nil, err
+	}
+	serverKey, err := os.ReadFile("assets/server.key")
+	if err != nil {
+		return nil, err
+	}
+	cert, err := tls.X509KeyPair(serverCert, serverKey)
+	if err != nil {
+		return nil, err
+	}
+	ts.TLS = &tls.Config{Certificates: []tls.Certificate{cert}}
+	ts.StartTLS()
+	return ts, nil
+}
+
+var _ = Describe("HelmOps controller", func() {
+	When("a new HelmApp is created", func() {
+		var helmapp fleet.HelmApp
+		var targets []fleet.BundleTarget
+		var doAfterNamespaceCreated func()
+		JustBeforeEach(func() {
+			os.Setenv("EXPERIMENTAL_HELM_OPS", "true")
+			nsSpec := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}
+			err := k8sClient.Create(ctx, nsSpec)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(k8sClient.Create(ctx, &helmapp)).ToNot(HaveOccurred())
+			if doAfterNamespaceCreated != nil {
+				doAfterNamespaceCreated()
+			}
+
+			DeferCleanup(func() {
+				Expect(k8sClient.Delete(ctx, nsSpec)).ToNot(HaveOccurred())
```

Comment:
nit: since not all test cases do it:
```suggestion
				Expect(k8sClient.Delete(ctx, nsSpec)).ToNot(HaveOccurred())
				_ = k8sClient.Delete(ctx, helmApp)
```

Reply:
Changed as suggested